updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -64,6 +72,23 @@ def save_job_details
                                        ))
     end
 
+    def dependency_files_for_multi_directories
+      all_dependency_files = job.source.directories.map do |dir|
+        updated_dir_source = job.source.clone.tap { |s| s.directory = dir }
+        ff = FileFetchers.for_package_manager(job.package_manager).new(
+          source: updated_dir_source,
+          credentials: Environment.job_definition.fetch("credentials", []),
+          options: job.experiments
+        )
+        ff.files
+      end
+      all_dependency_files.flatten
+    rescue Octokit::BadGateway
+      @file_fetcher_retries ||= 0
+      @file_fetcher_retries += 1
+      @file_fetcher_retries <= 2 ? retry : raise
```

Comment:
Does the rescue need to go inside the map? It looks like if it does throw BadGateway it will retry from the start.

Reply:
Thanks for the review, I will adjust the `dependency_files_for_multi_directories` method to retry fetching only the directory that caused the BadGateway error, rather than restarting from the beginning